dali/operators/reader/parser/tfrecord_parser.h

```diff
@@ -58,13 +58,9 @@ class TFRecordParser : public Parser<Tensor<CPUBackend>> {
 
     // Omit length and crc
     raw_data = raw_data + sizeof(length) + sizeof(crc);
-    try {
-      DALI_ENFORCE(example.ParseFromArray(raw_data, length),
-          "Error in parsing - invalid TFRecord file!");
-    } catch (std::exception& e) {
-      std::string str = "Error while parsing TFRecord: " + std::string(e.what());
-      DALI_FAIL(str);
-    }
+    DALI_ENFORCE(example.ParseFromArray(raw_data, length),
```

Comment:
Could you double check if `ParseFromArray` cannot throw and we don't need this try-catch here?

Reply:
The return code is used to signal an error. I was looking at the documentation and couldn't find anything about exceptions. Moreover, then I recalled that Google doesn't use C++ exceptions in their code: https://google.github.io/styleguide/cppguide.html#Exceptions

Reply:
Ok